src/js/sync.js

```diff
@@ -0,0 +1,93 @@
+var SailthruGigya = {
+
+  options : {},
+  profile : {},
+
+  init : function(options) {
+
+    if (!window.Sailthru) {
+      console.log('Sailthru JS is not loaded.Ensure you have included the JS on page');
+      return;
+    }
+
+    if (gigya) {
+        this.options = options;
+        gigya.socialize.addEventHandlers({
+          onLogin:this.syncProfile
+       });
+    } 
+  },
+
+  syncProfile : function(eventObj) {
+
+      // Allow customer to send this to server side too for additional processing.  
+      if (SailthruGigya.options.callback_url) {
+        SailthruGigya.processCallback(eventObj, SailthruGigya.options.callback_url);
+      }
+
+      var userKey = '';
+      var vars = {};
+      var lists = {};
+      var params = {};
+
+
+      if (eventObj.user.email !=='' || typeof eventObj.user.email != 'undefined') {
+       
+        // default vars to exclude
+        var exclude = 'UIDSig, UIDSignature, signatureTimestamp, capabilities, statusCode, statusReason, signatureTimestamp, isTempUser, isConnected, isLoggedIn, isSiteUID, isSiteUser, oldestDataUpdatedTimestamp';
+        
+        for (let key in eventObj.user) {        
+          
+          if (exclude.indexOf(key) < 0) {
```

Comment:
This isn't clear and doesn't include the exclude vars set in `options`.

You should break checking for exclude into it's own tiny function called `is_excluded_var(var_name)` (or something like it) which takes a var name and checks it against your default exclude list and the user-defined list and returns a boolean.

Reply:
e.g. something like
```
is_excluded(string var_name) {
   return !indexOf(var_name, this.excluded_vars) and !indexOf(var_name, default_excludes);
}

Reply:
@nickgundry is this part verified working? Did we run through this?